pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/AdditionWithPrecisionTransformFunction.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.common.function.scalar.DataTypeConversionFunctions;
+import org.apache.pinot.core.common.DataSource;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.core.plan.DocIdSetPlanNode;
+
+
+public class AdditionWithPrecisionTransformFunction extends BaseTransformFunction {
+
+  public static final String FUNCTION_NAME = "addWithPrecision";
+
+  private List<TransformFunction> _transformFunctions = new ArrayList<>();
+  private BigDecimal[] _sums;
+  private Integer _precision = null;
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    // Check that there are more than 1 arguments
+    if (arguments.size() < 3) {
+      throw new IllegalArgumentException("At least 3 arguments are required for ADD transform function");
+    }
+
+    for (TransformFunction argument : arguments) {
+      if (argument instanceof LiteralTransformFunction) {
```

Comment:
Parse the last argument as the precision instead of identifying `LITERAL` as precision? In certain cases user can add `LITERAL` to a column, e.g. `addWithPrecision(col, BigDecimal(123), 0)`

Reply:
Not applicable now.